wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php

```diff
@@ -126,32 +135,36 @@ private function generateData() {
 			}
 		}
 		
+		$contentProviders = ObjectTypeCache::getInstance()->getObjectTypes('com.woltlab.wcf.content.userContentProvider');
+		
+		// sort object types
+		uasort($contentProviders, function ($a, $b) {
+			$niceValueA = ($a->nicevalue ?: 0);
+			$niceValueB = ($b->nicevalue ?: 0);
+			
+			return $niceValueA <=> $niceValueB;
+		});
+		
 		foreach ($contentProviders as $contentProvider) {
 			if ($this->contentProvider === null || (is_array($this->contentProvider) && in_array($contentProvider->objectType, $this->contentProvider))) {
-				/** @var IUserContentProvider $processor */
-				$processor = $contentProvider->getProcessor();
-				$contentList = $processor->getContentListForUser($this->user);
-				$count = $contentList->countObjects();
-				
-				if ($count) {
-					$this->data['provider'][$contentProvider->objectType] = [
-						'count' => $count,
-						'objectTypeID' => $contentProvider->objectTypeID,
-						'nicevalue' => $contentProvider->nicevalue ?: 0
-					];
+				foreach ($this->users as $user) {
```

Comment:
I would invert the order of the 2 loops:

Outer loop: Users.
Inner Loop: Content Providers.

This ensures that users after processed one after another which is the more natural order IMO.

Reply:
I think in the end it doesn't matter. I want to avoid having to check the content providers several times. That would make the code less readable, because the user doesn't notice when and how what is deleted anyway.